tracee-rules/engine/engine.go

```diff
@@ -152,11 +154,31 @@ func (engine *Engine) consumeSources(done <-chan bool) {
 					engine.signaturesMutex.RUnlock()
 					continue
 				}
+
+				u, err := traceeEvt.ToUnstructured()
+				if err != nil {
+					engine.logger.Printf("unstructuring tracee event: %v", err)
```

Comment:
While I'm not sure of the original intentions of the existing code that has a RWLock for each signature, I believe we will need to release the read lock in the failure branches in this code.


```suggestion
					engine.logger.Printf("unstructuring tracee event: %v", err)
					engine.signaturesMutex.RUnlock()
```

Reply:
Also a good read since we're talking about performance: https://github.com/golang/go/issues/17973

Reply:
We can also look into contention / block / mutex profiles